package-res/resources/web/pentaho/data/Filter.js

```diff
@@ -32,6 +46,66 @@ define([
     }
   });
 
+  /**
+   * @name Filter
+   * @memberOf pentaho.data
+   * @class
+   * @amd pentaho/data/Filter
+   *
+   * @classdesc The `Filter` class provides factory methods for building the following filters:
+   *
+   * * {@link pentaho.data.filter.IsEqual}
+   * * {@link pentaho.data.filter.IsIn}
+   * * {@link pentaho.data.filter.Or}
+   * * {@link pentaho.data.filter.And}
+   * * {@link pentaho.data.filter.Not}
+   * * {@link pentaho.data.filter.RootFilter}
+   *
+   * @example
+   * <caption> Use the available `Filter` factories.</caption>
+   *
+   * require(["pentaho/data/Table", "pentaho/data/Filter"], function(Table, Filter) {
+   *   var data = new Table({
+   *     model: [
+   *       {name: "product", type: "string", label: "Product"},
+   *       {name: "sales", type: "number", label: "Sales"},
+   *       {name: "inStock", type: "boolean", label: "In Stock"}
+   *     ],
+   *     rows: [
+   *       {c: [{v: "A"}, {v: 12000}, {v: true}]},
+   *       {c: [{v: "B"}, {v: 6000}, {v: true}]},
+   *       {c: [{v: "C"}, {v: 12000}, {v: false}]},
+   *       {c: [{v: "D"}, {v: 1000}, {v: false}]},
+   *       {c: [{v: "E"}, {v: 2000}, {v: false}]},
+   *       {c: [{v: "F"}, {v: 3000}, {v: false}]},
+   *       {c: [{v: "G"}, {v: 4000}, {v: false}]}
+   *     ]
+   *   });
+   *
+   *   var sales12k = new Filter.IsEqual("sales", [12000]);
+   *   var productAB = new Filter.IsIn("product", ["A", "B"]);
+   *   var notInProductABFilter = new Not(productAB);
+   *   var andFilter = new Filter.And([sales12k, notInProductABFilter]);
+   *   var filter = new Not(andFilter);
+   *   var filteredData = filter.apply(data); //filteredData.getValue(0, 0) === "A", filteredData.getValue(1, 0) === "B", filteredData.getValue(2, 0) === "D", filteredData.getValue(3, 0) === "E", filteredData.getValue(4, 0) === "F", filteredData.getValue(5, 0) === "G"
+   *
+   *   //Or alternatively,
+   *   var spec = {
+   *      "$not": {
+   *        "$and": [
+   *          {"sales": 12000},
+   *          {"$not": {"product": {"$in": ["A", "B"]}}}
+   *        ]
+   *      }
+   *    };
+   *
+   *   var filterFromSpec = Filter.create(spec);
+   *   var filteredDataFromSpec = filterFromSpec.apply(data); //filteredDataFromSpec.getValue(0, 0) === "A", filteredDataFromSpec.getValue(1, 0) === "B", filteredDataFromSpec.getValue(2, 0) === "D", filteredDataFromSpec.getValue(3, 0) === "E", filteredDataFromSpec.getValue(4, 0) === "F", filteredDataFromSpec.getValue(5, 0) === "G"
+   * });
+   *
+   * @description The `Filter` class is a set of factories for building filters.
+   *
```

Comment:
This just repeats the @classdesc tag. If you can't come up with a better piece of text (say a more in-depth exploration on how to use the class, or in which circunstances it should be used or not), then just remove this line.

In a way, the example you added is an illustration of the contents @description